### PR TITLE
Added support for host-process container installation

### PR DESCRIPTION
**Pull request by song-jiang**
## Description

This PR is the first step for https://github.com/microsoft/ebpf-for-windows/issues/996 which adds scripts/dockerfile to build host-process container image. It also added manifests which can be deployed in a Kubernetes cluster. 

Note:
- The scripts need to be updated once the release process is finalised. 
- Which repository should the image been pushed?

## Testing

_Do any existing tests cover this change? Are new tests needed?_

- Need CI pipeline for building the images and deploy them. 
- Need CD pipeline to push the verified images to the repository.  

## Documentation

_Is there any documentation impact for this change?_
Yes. We would need update the doc on how to install ebpf-for-windows with host-process container.

### CODE REVIEW

File: manifests/Kubernetes/ebpf-for-windows-daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      tolerations:
+      - operator: Exists
+      securityContext:
+        windowsOptions:
+          hostProcess: true
+          runAsUserName: "NT AUTHORITY\\system"
+      hostNetwork: true
+      containers:
+      - name: ebpf-for-windows
+        image: docker.io/songtjiang/ebpfwin-install:01
```
Comment (song-jiang) [inline manifests/Kubernetes/ebpf-for-windows-daemonset.yaml:26]: This is just a test image. Will update it later.

**Review by dthaler — COMMENTED** [review]
@song-jiang Can you add an .md file to the docs directory that explains how to use?

**Review by dthaler — COMMENTED** [review]
Looks ok to me other than some spelling and grammatical nits.

**Comment by song-jiang** [comment]
Thanks @dthaler for review comments and suggestions.

**Review by dthaler — APPROVED** [review]

**Comment by song-jiang** [comment]
@dthaler Could you please take another look as I replaced my own test image path in the manifest file. Once the PR get approved, should I squash the commit and let you merge?

**Comment by dthaler** [comment]
> @dthaler Could you please take another look as I replaced my own test image path in the manifest file. Once the PR get approved, should I squash the commit and let you merge?

Will do.  You don't need to squash the commit, we use github's "squash and merge" functionality so it will be automatically squashed as part of the merge.

**Review by dthaler — DISMISSED** [review]

**Comment by song-jiang** [comment]
@dthaler `RELEASE_ZIP` argument was added in the wrong place after the Dockerfile passed my test.  I've pushed a new commit to fix this. Thanks.

**Review by dthaler — DISMISSED** [review]

**Review by dthaler — DISMISSED** [review]

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**